### PR TITLE
test: add test asserting loader headers can be set on response

### PR DESCRIPTION
**Pull request by pieh**
## Description

This adds user journey tests related to caching

## Related Tickets & Documents

Closes https://linear.app/netlify/issue/FRA-509/implement-remix-user-journey-tests-related-to-caching

Given | When | Then | Note
-- | -- | -- | --
loader returns custom Cache-Control: foo response header for A | load A | serves response with Cache-Control: foo header | Implemented
configured middleware via framework which sets custom Foo: bar response header for A | load A | serves response with Foo: bar header | Not applicable / not implemented
user-defined Netlify Edge Function middleware which sets custom Foo: bar response header for A | load A | serves response with Foo: bar header | Implemented
configured SWR=90 for A | load Await 5sload Await 86sload Await 5sload A* requires sticky CDN node targeting | serves response from origin function (v1)serves pre-rendered response from CDN edge (v1)serves pre-rendered response from CDN edge (v1)serves pre-rendered response from CDN edge (v2) | Implemented
configured SWR=90 for A | load Await 5sload Ainvalidate A on demandload A | serves response from origin function (v1)serves pre-rendered response from CDN edge (v1)serves response from origin (v2) | Implemented

Implemented tests were implemented for Vite Serverless and Vite Edge cases

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes_

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/remix-compute/issues/new/choose) before writing your code 🧑‍💻. This
      ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a
      typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 📖. This ensures your code follows our style
      guide and passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

### CODE REVIEW

File: tests/e2e/user-journeys.spec.ts [hunk 1]
```diff
@@ -119,4 +119,146 @@ test.describe('User journeys', () => {
...
+    await expect(page.getByRole('heading', { name: /Headers/i })).toBeVisible()
+    expect(response?.headers()['cache-control']).toBe('public,max-age=3600')
+  })
+
+  test('user can configure Stale-while-revalidate when using origin SSR', async ({ page, serverlessSite }) => {
+    // attempt to purge cdn cache in case this is a retry
+    await fetch(`${serverlessSite.url}/purge-cdn?tag=stale-while-revalidate-tag`)
+
+    await new Promise((resolve) => setTimeout(resolve, 1000))
+
```
Comment (serhalp) [inline tests/e2e/user-journeys.spec.ts:140]:
```suggestion
```

I [don't think this should be necessary](https://playwright.dev/docs/test-retries#failures):

> Should any test fail, Playwright Test will discard the entire worker process along with the browser and will start a new one. Testing will continue in the new worker process starting with the next test.

Since it discards the worker and retries in a new one, and fixtures are scoped to workers, a new fixture will be created and deployed.

File: tests/e2e/user-journeys.spec.ts [hunk 2]
```diff
@@ -119,4 +119,146 @@ test.describe('User journeys', () => {
...
+    await page.goto(`${serverlessSite.url}/stale-while-revalidate`)
+    const responseGeneratedAtText1 = await page.getByText('Response generated at').textContent()
+
+    await new Promise((resolve) => setTimeout(resolve, 5000))
```
Comment (serhalp) [inline tests/e2e/user-journeys.spec.ts:144]: We can replace all of these with a built-in:
```suggestion
    await page.waitForTimeout(5000);
```

File: tests/e2e/user-journeys.spec.ts [hunk 3]
```diff
@@ -119,4 +119,146 @@ test.describe('User journeys', () => {
...
+
+    await page.goto(`${serverlessSite.url}/stale-while-revalidate`)
+    const responseGeneratedAtText2 = await page.getByText('Response generated at').textContent()
+    expect(responseGeneratedAtText2, 'First and second response should have matching date and time').toEqual(
+      responseGeneratedAtText1,
+    )
```
Comment (serhalp) [inline tests/e2e/user-journeys.spec.ts:150]: 👍🏼 Nice

File: tests/e2e/user-journeys.spec.ts [hunk 4]
```diff
@@ -119,4 +119,146 @@ test.describe('User journeys', () => {
...
+      'Fourth response should not have matching date and time with previous responses',
+    ).not.toEqual(responseGeneratedAtText1)
+  })
+
+  test('user can configure Stale-while-revalidate when using edge SSR', async ({ page, edgeSite }) => {
+    // attempt to purge cdn cache in case this is a retry
+    await fetch(`${edgeSite.url}/purge-cdn?tag=stale-while-revalidate-tag`)
+
+    await new Promise((resolve) => setTimeout(resolve, 1000))
+
```
Comment (serhalp) [inline tests/e2e/user-journeys.spec.ts:175]:
```suggestion
    // attempt to purge cdn cache in case this is a retry
    await fetch(`${edgeSite.url}/purge-cdn?tag=stale-while-revalidate-tag`)

    await new Promise((resolve) => setTimeout(resolve, 1000))

```

**Review by serhalp — DISMISSED** [review]
Looks great!

**Review by serhalp — APPROVED** [review]

File: tests/e2e/user-journeys.spec.ts (hunk 1 above)
Comment (pieh) [reply tests/e2e/user-journeys.spec.ts]: Ah that's good to know. I did temporarily hardcode fixture url when iterating on test, so I was getting same deploy and for that I did need purge for reruns, but definitely not needed when new deploys happen on test failures (and fact that we don't use that page / pathname in other tests)
